agent.go

```diff
@@ -875,6 +892,71 @@ func (s *sandbox) setupSignalHandler() error {
 	return <-errCh
 }
 
+type exitStatusList []*exitProcessStatus
+
+func (p exitStatusList) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+func (p exitStatusList) Len() int           { return len(p) }
+func (p exitStatusList) Less(i, j int) bool { return p[i].sequenceNum-p[j].sequenceNum < 0 }
+
+func (e *exitProcess) add(id string, exitCode chan int) {
+	e.Lock()
+	defer e.Unlock()
+
+	if e.exitCodes == nil {
+		e.exitCodes = map[string]*exitProcessStatus{}
+	}
+
+	// delete (first in and first out) half of the elements if the length arrive maxExitCodeNum
+	if len(e.exitCodes) >= maxExitCodeNum {
```

Comment:
this means it couldn't support more than maxExitCodeNum execs run and exit at the same time?  
This isn't what we want.